File: backend/aws-api/tree-crud/PostController.py
import json

def post(connection, data):
    cursor = connection.cursor()
    body = json.loads(data)
    
    print("=====testing=====")
    print(body)
    
    sql = "INSERT INTO trees "
    sql += "(tree_name_en, tree_name_cn, tree_alias, scientific_name, family_code, ecologic, cap_96, cap_586, hk_rare, cn_rare, flowering, fruit, tree_desc_en, tree_desc_cn, tree_image) "
    sql += "VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)"
    val = (
        body["treeNameEn"],
        body["treeNameCn"],
        body["treeAlias"],
        body["scientificName"],
        body["familyCode"],
        body["ecologic"],
        
        body["cap96"],
        body["cap586"],
        body["hkRare"],
        body["cnRare"],
        
        body["flowering"],
        body["fruit"],
        
        body["treeDescEn"],
        body["treeDescCn"],
        body["treeImage"]
        
    )
    
    statusCode = 502
    message = "Create tree record failure"
    try:
        cursor.execute(sql, val)
        connection.commit()
        statusCode = 200
        message = "Create tree record succeeded "
    except:
        connection.rollback()
    
    return {
        'statusCode': statusCode,
        'headers': {
            'Content-Type': 'application/json',
            'Access-Control-Allow-Headers': 'Content-Type',
            'Access-Control-Allow-Origin': '*',
            'Access-Control-Allow-Methods': 'OPTIONS,POST'
        },
        'body': message
    }
    

File: backend/aws-api/user-crud/PostController.js
const { decrypt, hash } = require("CryptoUtil");

exports.post = function(pool, body, callback) {
    let obj = JSON.parse(body),
        userName = obj["userName"],
        email = obj["email"],
        membership = obj["membership"],
        password = obj["password"],
        iv = obj["iv"],
        key = process.env.PASSWORD_KEY,
        decrypted = decrypt(password, key, iv),
        hashed = hash(decrypted),
        sql = "INSERT INTO users (user_name, email, password, membership) VALUES ";
    
    sql += `('${userName}', '${email}', '${hashed}', ${membership});`;
        
    pool.getConnection(function(err, connection) {
        if(err) throw err;
        connection.query(sql, function(error, results) {
            if(error) {
                callback(error);
            } else {
                const result = {
                    'statusCode': 200,
                    'headers': {
                        'Content-Type': 'application/json',
                        'Access-Control-Allow-Headers': 'Content-Type',
                        'Access-Control-Allow-Origin': '*',
                        'Access-Control-Allow-Methods': 'OPTIONS,PUT'
                    },
                    'body': "Create user success"
                };
                callback(null, result);
            }
        });
    });
};